src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -223,17 +216,20 @@ export class SavedObjectsClient {
    * @returns {promise}
    */
   async update(type, id, attributes, options = {}) {
-    const response = await this._withKibanaIndexAndMappingFallback('update', {
-      id,
-      type,
+    // handles documents with an _id of `${type}:${id}`
+    const docResponse = await this._withKibanaIndex('search', { body: createIdQuery({ type, id }) });
```

Comment:
When a document is created in 5.x with multi-types, we create the document with the id provided.
When we re-index, we are pre-pending the type to the id due to uniqueness only being guaranteed per-type.
Currently, we are utilizing Elastisearch to provide the unique ID's and if we attempt to create a document with a specific ID, then we continue to prepend it.

I believe we should change using `uuid.v1()` or similar and auto-generate the ids ourselves. This would allow us to, beginning in 6.0, to do lookups/updates/gets by ID.

The change here fixes the issue and I am working to implement and test the above.